### PR TITLE
task timeline visualization

### PR DESCRIPTION
**Pull request by danielnorberg**
An attempt to implement a basic gantt-like task execution timeline visualization.

screen shot 2016-12-13 at 5 49 29 pm

### CODE REVIEW

File: digdag-ui/console.jsx [hunk 1]
```diff
@@ -1226,6 +1226,55 @@ const TaskListView = (props:{tasks: Array<Task>}) =>
...
+          <th>Name</th>
+          <th>Timeline</th>
+        </tr>
+      </thead>
+      <tbody>
+        {
+          props.tasks.map(task => {
+            var startTime = props.startTime
+            var endTime = props.endTime
+            var style = {}
```
Comment (dzannotti) [inline digdag-ui/console.jsx]: `let style`

**Comment by dzannotti** [comment]
This branch has a bunch of linting errors atm (`npm run lint`)

File: digdag-ui/console.jsx [hunk 2]
```diff
@@ -1270,6 +1319,78 @@ class AttemptTasksView extends React.Component {
...
+  tasks.forEach(task => {
+    if (task.startedAt === null) {
+      return
+    }
+    var startedAt = moment(task.startedAt)
+    if (first === null || startedAt.isBefore(first)) {
+      first = startedAt
+    }
+  })
+  return first
```
Comment (cesarandreu) [inline digdag-ui/console.jsx]: If you don't mind the extra cost, FP-style JS is nice (and IMO easier to read):

```js
return tasks
  .filter(task => task.startedAt !== null)
  .map(task => moment(task.startedAt))
  .reduce((first, startedAt) => first === null || startedAt.isBefore(first) ? startedAt : first, null)
```

File: digdag-ui/console.jsx [hunk 3]
```diff
@@ -1226,6 +1226,55 @@ const TaskListView = (props:{tasks: Array<Task>}) =>
...
+      <thead>
+        <tr>
+          <th>Name</th>
+          <th>Timeline</th>
+        </tr>
+      </thead>
+      <tbody>
+        {
+          props.tasks.map(task => {
+            var startTime = props.startTime
```
Comment (danielnorberg) [inline digdag-ui/console.jsx]: 👍

File: digdag-ui/console.jsx (hunk 1 above)
Comment (danielnorberg) [inline digdag-ui/console.jsx]: 👍

File: digdag-ui/console.jsx (hunk 2 above)
Comment (danielnorberg) [inline digdag-ui/console.jsx]: 👍

**Comment by frsyuki** [comment]
Cool!

**Comment by danielnorberg** [comment]
Now also hides the root task (and removes the root task name prefix from task names)

**Pull request merged**